### PR TITLE
Documentation - Docusaurus port

### PR DESCRIPTION
**Pull request by InfraredAces**
This port should contain all of the current documentation that is live on the [documentation site](https://gp2040-ce.info/#/) as of the time of this PR.

I have a test version of this site running one https://infraredaces.github.io/docusaurus-test/ , if you would like to see it live and test.

There are only a few things left to do before and after this PR is merged to allow the workflow to deploy the site, but they are only performable by maintainers of the repo.

- Before: Create an access token with `Read and Write` access to the Contents of the repo, name it "ACTION_SECRET", and then place it in Repository Secrets. The name can be changed, however the deploy.yml contents will need to be updated as well.

image

- After: Change the target of the Github Pages to the generated `gh-pages` branch that the `deploy` action creates. 

image

This action will create a branch of that consists of only the static files generated during the build process and result in the pages seen in the test site linked above.

image

### CODE REVIEW

**Review by arntsonl — COMMENTED** [review]
Going to wait for at least one other review from a web-design stand-point, but I commented on things that stood out to me.

File: site/docs/development/firmware-development.mdx
Comment (InfraredAces) [reply site/docs/development/firmware-development.mdx:46]: Having the Tabs and TabItems properly indented results in an error due to the lack of closing tag, which are clearly present in the document. By having all the Tags unindented, the issue is resolved. Unsure of how to proceed.

image

File: site/.gitignore
Comment (InfraredAces) [reply site/.gitignore:37]: These are not specific to my setup, this is the .gitignore used by Docusaurus. `website` is the default name of the folder when generating a site using a template. Will remove unnecessary references.

https://github.com/facebook/docusaurus/blob/main/.gitignore

File: site/README.md
Comment (InfraredAces) [reply site/README.md:1]: Removed Docusaurus README

File: site/docusaurus.config.js
Comment (InfraredAces) [reply site/docusaurus.config.js:96]: Changed from "Docs" to "Home"

File: site/docusaurus.config.js
Comment (InfraredAces) [reply site/docusaurus.config.js:83]: Removed built in comments from Docusaurus Template generation.

File: site/docs/development/firmware-development.mdx
Comment (InfraredAces) [reply site/docs/development/firmware-development.mdx:46]: This seems to be an issue with interoperability between Markdown and JSX.

https://docusaurus.io/docs/markdown-features/react#markdown-and-jsx-interoperability

File: README.md
Comment (InfraredAces) [reply README.md:27]: Indeed, will remove

File: site/docs/development/firmware-development.mdx
Comment (arntsonl) [reply site/docs/development/firmware-development.mdx:46]: Ahhh okay, yeah so a technical limitation! No worries

**Review by arntsonl — COMMENTED** [review]
Nice! I'll wait for another reviewer until we merge but changes look good!

File: configs/Pico/BoardConfig.h
Comment (arntsonl) [inline configs/Pico/BoardConfig.h:320]: Looks like a bad '/' got merged in here

File: site/docs/assets/boards/ReflexEncoder.jpg
Comment (arntsonl) [inline site/docs/assets/boards/ReflexEncoder.jpg:1]: Did we replace this with V1/V2?

**Review by arntsonl — APPROVED** [review]
Looks really good!

**Comment by Pelsin** [comment]
Looks great!
Awesome work!

File: site/docs/assets/boards/ReflexEncoder.jpg
Comment (InfraredAces) [reply site/docs/assets/boards/ReflexEncoder.jpg:1]: This board has been replaced with ReflexEncode_v1.2.jpg and ReflexEncode_v2.0.jpg so we should remove

**Pull request merged**
